### PR TITLE
CI: Update first-party GitHub Actions from v3 to v4

### PR DESCRIPTION
**Pull request by RytoEX**
### Description

GitHub Actions has deprecated Actions based on node16. The v4 actions are based on node20. Replace first-party v3 actions with their v4 counterparts.

See:
https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/

### Motivation and Context

Want to resolve deprecation warnings on CI. Want to avoid CI failing when the node16 actions are disabled.

### How Has This Been Tested?

obs-plugintemplate, obs-deps, and obs-browser have both had successful CI runs. This repo's CI will also verify.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [ ] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: .github/workflows/push.yaml [hunk 1]
```diff
@@ -189,7 +189,7 @@ jobs:
           customLink: 'https://obsproject.com/'
 
       - name: Upload Artifacts 📡
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
```
Comment (derrod) [inline .github/workflows/push.yaml:192]: Right now the sparkle artifact relies on the fact that v3 artifacts are combined server side go produce a single output with both architectures. So just switching to v4 will not work.

Comment (RytoEX) [reply .github/workflows/push.yaml:192]: I can revert this specific change for now, but we'll want to address that "soon", per GitHub's current plan to "transition all actions to run on Node 20 by Spring 2024".

Comment (RytoEX) [reply .github/workflows/push.yaml]: Reverted this line.

Comment (derrod) [reply .github/workflows/push.yaml]: Yeah that's fine, we can probably just split them up as well, having them be one single artifact was convenient but not technically necessary.

**Pull request merged**